state-chain/amm/src/range_orders.rs

```diff
@@ -299,9 +297,9 @@ impl SwapDirection for BaseToQuote {
 		amount: Amount,
 	) -> SqrtPriceQ64F96 {
 		assert!(0 < liquidity);
-		assert!(SqrtPriceQ64F96::zero() < sqrt_price_current);
+		assert!(U256::zero() < sqrt_price_current.as_raw());
```

Comment:
Is it worth implementing operators like <, > for Price and SqrtPriceQ64F96? Alternatively, in this particular case you can just check that the price is not zero (by adding `is_zero()`).